public/kibi_timeline.js

```diff
@@ -203,68 +203,87 @@ define(function (require) {
               startRawFieldValue = hit.fields[params.startField];
 
               let endFieldValue = null;
+              if (params.endField) {
+                if (params.endFieldSequence) { // in kibi, we have the path property of a field
+                  endFieldValue = kibiUtils.getValuesAtPath(hit._source, params.endFieldSequence);
+                } else {
+                  endFieldValue = _.get(hit._source, params.endField);
+                }
+                endRawFieldValue = hit.fields[params.endField];
 
-              if (startFieldValue && (!_.isArray(startFieldValue) || startFieldValue.length)) {
-                if (timelineHelper.isMultivalued(startFieldValue)) {
-                  detectedMultivaluedStart = true;
+                if (endFieldValue.length !== startFieldValue.length) {
+                  notify.warning('Check your data - the number of values in the \'Event end date\'' +
+                  ' must be equal to the number of values in the \'Event start date\'');
+                  return;
                 }
+              }
+
+              if (startFieldValue && (!_.isArray(startFieldValue) || startFieldValue.length)) {
                 let indexId = searchSource.get('index').id;
-                let startValue = timelineHelper.pickFirstIfMultivalued(startFieldValue);
-                let startRawValue = timelineHelper.pickFirstIfMultivalued(startRawFieldValue);
-                let content =
-                  '<div title="index: ' + indexId +
-                  ', startField: ' + params.startField +
-                  (params.endField ? ', endField: ' + params.endField : '') +
-                  '">' + labelValue +
-                  (params.useHighlight ? '<p class="tiny-txt">' + timelineHelper.pluckHighlights(hit, highlightTags) + '</p>' : '') +
-                  '</div>';
-
-                let e =  {
-                  index: indexId,
-                  content: content,
-                  value: labelValue,
-                  start: new Date(startRawValue),
-                  startField: {
-                    name: params.startField,
-                    value: startValue
-                  },
-                  type: 'box',
-                  group: $scope.groupsOnSeparateLevels === true ? index : 0,
-                  style: 'background-color: ' + groupColor + '; color: #fff;',
-                  groupId: groupId
-                };
-
-                if (params.endField) {
-                  if (params.endFieldSequence) { // in kibi, we have the path property of a field
-                    endFieldValue = kibiUtils.getValuesAtPath(hit._source, params.endFieldSequence);
+
+                startFieldValue.forEach(function (value, i) {
```

Comment:
since `startFieldValue` may not be an array, either make sure it is or review this code.